server/prompt.go

```diff
@@ -44,9 +44,8 @@ func chatPrompt(ctx context.Context, m *Model, tokenize tokenizeFunc, opts *api.
 			return "", nil, errTooManyImages
 		}
 
-		// always include the last message
-		if i == n {
-			continue
+		if i == 0 && msgs[i].Role == "system" {
+			break
```

Comment:
This is assuming that we only need to preserve 1 system message and that it is the first message. However, I think the original intention of this function is to preserve all system messages.

Reply:
The current truncation strategy is as follows: when msgs is greater than 2, the system retains all system messages; when msgs equals 2, the first system message is removed when the truncation mechanism is triggered.
`
                system = make([]api.Message, 0)
		for j := range i {
			if msgs[j].Role == "system" {
				system = append(system, msgs[j])
			}
		}

		// always include the last message
		if i == n {
			continue
		}
`
The updated strategy is: when msgs equals 2, the first system message will be retained when the truncation mechanism is triggered.

Reply:
Actually, this change simply fixes the issue to ensure that when msgs equals 2, the truncation mechanism retains the first system message.

Reply:
What if the sequence of messages is user, system, user or system, system, system, user? With this change, it is not consistent in the behavior of system messages that are preserved.

As I said, I think the original goal was to preserve all system messages.